task/upload.go

```diff
@@ -738,14 +740,26 @@ func clipTaskOutputLocations(tctx *TaskContext) ([]OutputName, []clients.OutputL
 	return outputNames, outputLocations, nil
 }
 
+// out is a helper to create an output, to avoid repeated struct attribute names in the code
+func out(enabled, path string) output {
+	return output{
+		enabled: enabled,
+		path:    path,
+	}
+}
+
+// output represents each requested media output for outputLocations()
+type output struct {
+	enabled string
+	path    string
+}
+
 func outputLocations(
-	outURL,
+	outURL string,
 	hls,
-	hlsRelPath,
 	mp4,
-	mp4RelPath,
 	fmp4,
-	fmp4RelPath string,
+	thumbnails output,
```

Comment:
WDYT of using `*output` so we can use `nil` as the "not configured" value, instead of an empty struct?

Reply:
I wanted to avoid pointers so that nil checks aren't necessary etc but I've introduced a "holder" struct which means no need to explicitly type out the empty struct def. Hope that looks good to you

Reply:
Looks great!